server/src/main/java/io/crate/planner/operators/Rename.java

```diff
@@ -101,8 +102,8 @@ public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep)
         /* In `SELECT * FROM (SELECT t1.*, t2.* FROM tbl AS t1, tbl AS t2) AS tjoin`
          * The `ScopedSymbol`s are ambiguous; To map them correctly this uses a IdentityHashMap
          */
-        IdentityHashMap<Symbol, Symbol> parentToChildMap = new IdentityHashMap<>(outputs.size());
-        IdentityHashMap<Symbol, Symbol> childToParentMap = new IdentityHashMap<>(outputs.size());
+        HashMap<Symbol, Symbol> parentToChildMap = new HashMap<>(outputs.size());
+        HashMap<Symbol, Symbol> childToParentMap = new HashMap<>(outputs.size());
```

Comment:
I tried this before and think we cannot do it, see code comment above, it's needed to handle ScopedSymbol.

See https://github.com/crate/crate/pull/9745#discussion_r390234876.

Another idea was to enrich `ScopedSymbol` with even more context but that's something we tried for another fix and decided not to go with it. See https://github.com/crate/crate/pull/14816#discussion_r1370991424